src/types/pointer.md

```diff
@@ -50,6 +50,12 @@ Raw pointers can be created directly using [`core::ptr::addr_of!`] for `*const`
 
 The standard library contains additional 'smart pointer' types beyond references and raw pointers.
 
+## Bit validity
+
+Despite pointers and references being similar to `usize`s in the machine code emitted on most platforms,
+the semantics of transmuting a reference or pointer type to a non-pointer type is currently undecided.
+Thus, it may not be valid to transmute a pointer or reference type, `P`, to a `[u8; size_of::<P>()]`.
```

Comment:
```suggestion
Thus, it may not be valid to transmute a pointer or reference type, `P`, to a `[u8; size_of::<P>()]`.

For thin raw pointers (i.e., for `T: Sized` and `P = *const T` or `P = *mut T`), the inverse direction is always valid.
However, the resulting pointer may not be dereferenced (not even if the pointee type has size zero).
```
Note however that there are thin pointers where `T: !Sized` (due to extern types). So maybe instead of `T: Sized` we should say `<T as Pointee>::Metadata == ()`.

Reply:
Sounds good, but do we want to refer to unstable features (the ptr_metadata feature for `Pointee`) in the reference? Maybe we could instead say `size_of::<P>() == size_of::<usize>()`?

Reply:
IMO using `Metadata` is more clear even if it isn't stable; I will defer to @ehuss here.

Reply:
@ehuss Any preference on the wording here? I'm happy to go with whatever you recommend.

Reply:
We generally try to avoid referring to any unstable elements in the Reference, so if you can avoid it, it would be appreciated. If @RalfJung is ok with that wording as a compromise, it seems good to me.

Unfortunately "thin" and "fat" pointers currently aren't defined in the reference, so it is a little awkward to start using terms that aren't defined. I don't think that necessarily needs to be addressed in this PR, but it is something to be aware of (and eventually resolved).

Reply:
Sounds good, updated. Should be ready to go!